docs/tutorials/runtime-module-in-own-crate.md

```diff
@@ -0,0 +1,167 @@
+---
+title: "Write a Runtime Module as a Rust crate"
+---
+
+In this tutorial, you'll write a Substrate runtime module that lives in its own crate, and include it in a node based on the node-template. If you prefer you can follow along with the [video walkthrough](todo record after initial feedback).
+
+## Setup a Development Environment
+If you haven't already, follow the guide to [install substrate](getting-started/installing-substrate.md). You'll need the substrate-up scripts, but not the standard substrate node, so the `--fast` option is what you want here.
+
+## Create your node
+We'll begin by creating a brand new node template.
+```bash
+substrate-node-new modular-chain <your name>
+```
+If you've followed other runtime tutorials, you've probably started this way.
+
+In this tutorial, we're not going to write our module directly as part of the node template, but rather as a separate Rust crate. This approach provides a few advantages:
+1. Our module can be easily imported into other nodes in the future.
+2. Our module can be published to places like github or crates.io without publishing our entire node in the same place.
+
+We do still need the node template, however, to actually run our module.
+
+## Clone the Module Template Repository
+So if our module isn't going to live in the node template, where _is_ it going to live? If you aren't familiar with library crates or how to use them in other rust projects, it's may be worth reading the [cargo docs](https://doc.rust-lang.org/cargo/guide/creating-a-new-project.html).
+
+While it's perfectly possible to create a new crate from scratch, the fastest way to get going with a new module is to clone the template repository.
+
+```bash
+git clone https://github.com/shawntabrizi/substrate-module-template
+```
+
+You should be able to immediately compile this template module with
+```bash
+cargo build --release
+```
+
+But before we jump straight to including our module in a runtime, let's take a look at the `Cargo.toml` file.
+
+After the standard naming and authorship information, we see a few lines about the `std` feature. Because substrate runtime code targets wasm, all the dependencies we use must be able to compile with `no_std`. Our `Cargo.toml` file begins by telling dependencies to only use their `std` feature when this module also uses its `std` feature. You will need to add each dependency you include here.
+
+```toml
+[features]
+default = ['std']
+std = [
+    'parity-codec/std',
+    'support/std',
+    'system/std',
+]
+```
+Our module will depend on low level features such as `system` and `support` from substrate itself. It will also be depended upon by a substrate-based runtime from above. So some care may be needed to ensure version compatibility on both sides. Substrate does not yet have releases published on crates.io, so instead we'll use git-based versioning.
```

Comment:
```suggestion
Our module will depend on low level Substrate module libraries such as `system` and `support`. It will also be a dependency by the Substrate-based runtime from above. So some care may be needed to ensure version compatibility on both sides. Substrate does not yet have releases published on crates.io, so instead we need to use git-based versioning.
```